src/intra_edge.rs

```diff
@@ -164,62 +164,27 @@ impl EdgeBranch {
         ];
         let node = EdgeNode { o, h, v };
 
-        let h4 = [
-            edge_flags.union(EdgeFlags::LEFT_HAS_BOTTOM),
-            EdgeFlags::LEFT_HAS_BOTTOM.union(
-                edge_flags
-                    .intersection(EdgeFlags::I420_TOP_HAS_RIGHT)
-                    .select(matches!(bl, BlockLevel::Bl16x16)),
-            ),
-            EdgeFlags::LEFT_HAS_BOTTOM,
-            edge_flags.intersection(EdgeFlags::LEFT_HAS_BOTTOM),
-        ];
-
-        let v4 = [
-            edge_flags.union(EdgeFlags::TOP_HAS_RIGHT),
-            EdgeFlags::TOP_HAS_RIGHT.union(
-                edge_flags
-                    .intersection(EdgeFlags::union_all([
-                        EdgeFlags::I420_LEFT_HAS_BOTTOM,
-                        EdgeFlags::I422_LEFT_HAS_BOTTOM,
-                    ]))
-                    .select(matches!(bl, BlockLevel::Bl16x16)),
-            ),
-            EdgeFlags::TOP_HAS_RIGHT,
-            edge_flags.intersection(EdgeFlags::TOP_HAS_RIGHT),
-        ];
+        let h4 = EdgeFlags::LEFT_HAS_BOTTOM.union(
+            edge_flags
+                .intersection(EdgeFlags::I420_TOP_HAS_RIGHT)
+                .select(matches!(bl, BlockLevel::Bl16x16)),
+        );
```

Comment:
the convention we went with here is that variables can use `union`, so `foo.union(...)`, but when the left-hand side is an enum variant we use `union_all`:

```suggestion
        let h4 = EdgeFlags::union_all([
            EdgeFlags::LEFT_HAS_BOTTOM,
            edge_flags
                .intersection(EdgeFlags::I420_TOP_HAS_RIGHT)
                .select(matches!(bl, BlockLevel::Bl16x16)),
        ]);
```